### ISSUE TITLE
optimizing threshold based on validation set

### ISSUE THREAD

**Issue by alongnn** [UKPLab/sentence-transformers#755]
When using BinaryClassificationEvaluator.py, the threshold is optimized w.r.t. the test set. It should be optimized w.r.t. the validation set since the labels of the test set are not available at this point.

**Comment by nreimers** [comment]
Yes, this is right. You can update your evaluator so that a fixed (pre-defined) threshold is used.

**Comment by alongnn** [comment]
Doesn’t it affect your results?

**Comment by nreimers** [comment]
The models here were evaluated with other evaluators.

**Comment by alongnn** [comment]
thanks
